torch/csrc/autograd/variable.cpp

```diff
@@ -509,7 +510,10 @@ void handle_view_on_rebase(DifferentiableViewMeta* diff_view_meta, bool indirect
     if (grad_fn) {
       msg = c10::str("Output ", diff_view_meta->output_nr_, " of ", grad_fn->name(), " is a view and ",
                      modified_obj, " modified inplace.");
+    } else if (creation_meta == CreationMeta::INFERENCE_MODE) {
+      msg = c10::str("A view was created in inference mode and ", modified_obj, " modified inplace in normal mode.");
     } else {
+      TORCH_INTERNAL_ASSERT_DEBUG_ONLY(creation_meta == CreationMeta::NO_GRAD_MODE);
```

Comment:
Actually I am not sure this will be accurate if you do (the old code had the same issue and was not reporting the right error message AFAIU):
```
a = torch.rand(10)
b = a.view_as(a)
b.requires_grad_()
```

The creation meta here will be default but you don't have a grad_fn.

Reply:
I removed this line but the example actually doesn't hit here, it's blocked by https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/variable.cpp#L413
and we won't hit this line if creation_meta=Default.